### PR TITLE
fix: 🤔 z-index in sd-select

### LINKED ISSUE #534: fix: 🤔 z-index sd dropdown
## Current behavior
If you place an sd-button under a sd-dropwdown the button is above the dropdown if you open it.
image

Not sure if this should be the behaviour or not -> open for discussion.

## Expected behavior
Dropdown is above button?

## Steps to reproduce
image

## DoR
- [ ] Item has business value
- [ ] Item has been estimated by the team
- [ ] Item is clear and well-defined
- [ ] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by azraefendic**
## Description:
Added a z-index on ::part(popup) in sd-select. 

Closes #534 

## Definition of Reviewable:
*PR notes: Irrelevant elements should be removed.*
- [x] relevant tickets are linked
- [x] PR is assigned to project board

### CODE REVIEW

**Comment by mariohamann** [comment]
Could you please add a play story for that and test it on Chromatic to make sure that this bug won't come back? I removed `skip chromatic` as this is definitely component relevant.

**Comment by mariohamann** [comment]
And I'm a bit unsure: https://github.com/solid-design-system/solid/issues/534#issue-1987490056

This only fixes the select, but not the dropdown, doesn't it? I expected that something in `sd-popup` is missing.

File: packages/components/src/components/select/select.ts [hunk 1]
```diff
@@ -1041,7 +1041,7 @@ export default class SdSelect extends SolidElement implements SolidFormControl {
       }
 
       sd-popup::part(popup) {
-        @apply overflow-y-scroll;
+        @apply overflow-y-scroll z-dropdown;
```
Comment (saemik94) [inline packages/components/src/components/select/select.ts:1044]: Just wondering what z-dropdown does, due to the fact that I'm not able to find any documentation about it 🤔

Comment (azraefendic) [reply packages/components/src/components/select/select.ts:1044]: You can find it in **_create-theme.cjs_**:
Screenshot 2024-01-17 at 14 24 01

Comment (saemik94) [reply packages/components/src/components/select/select.ts:1044]: thx

**Review by saemik94 — APPROVED** [review]

**Comment by azraefendic** [comment]
@mariohamann so at the time the ticket was created we were missing the z-index tokens, which were added in the meantime and I already tested the dropdown with Chris with just a later version and it was fixed since the class was already there.
And I will add another story then to test it on Chromatic.

**Review by yoezlem — APPROVED** [review]

**Review by mariohamann — APPROVED** [review]

**Comment by mariohamann** [comment]
> @mariohamann so at the time the ticket was created we were missing the z-index tokens, which were added in the meantime and I already tested the dropdown with Chris with just a later version and it was fixed since the class was already there. And I will add another story then to test it on Chromatic.

Okay, thanks for the explanation, sounds good! Story would be cool, but I would be fine even without one. :)

**Pull request merged**